rclcpp/include/rclcpp/any_subscription_callback.hpp

```diff
@@ -35,17 +35,23 @@ struct AnySubscriptionCallback
   using SharedPtrCallback = std::function<void(const std::shared_ptr<MessageT> &)>;
   using SharedPtrWithInfoCallback =
       std::function<void(const std::shared_ptr<MessageT> &, const rmw_message_info_t &)>;
+  using ConstSharedPtrCallback = std::function<void(const std::shared_ptr<const MessageT> &)>;
```

Comment:
indeed. Well, I'll do some refactoring to see if I can make the correct callback signature work in rclcpp in this branch.